perfkitbenchmarker/providers/yandexcloud/util.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2020 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for working with Yandex.Cloud resources."""
+
+import collections
+import json
+import yaml
+import logging
+import re
+from perfkitbenchmarker import context
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import virtual_machine
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.providers.yandexcloud import yc_machine_types
+import six
+
+if six.PY2:
+  import functools32 as functools
+else:
+  import functools
+
+FLAGS = flags.FLAGS
+FLAVORS = yc_machine_types.FLAVORS
+
+
+@functools.lru_cache()
+def GetDefaultFolderId():
+  """Get the default folder id."""
+  conf = ParseConfig()
+  if "folder-id" in conf:
+      return conf["folder-id"]
+
+  return None
+
+
+def GetDefaultZone():
+  """Get the default zone."""
+  conf = ParseConfig()
+  if "compute-default-zone" in conf:
+      return conf["compute-default-zone"]
+
+  return None
+
+
+@functools.lru_cache()
+def ParseConfig():
+  """Parse yc config file"""
+  cmd = [FLAGS.yc_path, 'config', 'list']
+  stdout, _, _ = vm_util.IssueCommand(cmd)
+
+  try:
+    return yaml.safe_load(stdout)
+  except yaml.YAMLError:
+    return {}
+
+
+def ReturnFlavor(machine_type):
+    """Returns RAM and Core values based on machine_type selection."""
+
+    logging.info('Fetching flavor specs for new VM.')
+    for flavor in FLAVORS:
+        if(machine_type == flavor['name']):
+            return flavor['memory'], flavor['cores']
+
+
+class YcCommand(object):
+  """A yc command.
+
+  Attributes:
+    args: list of strings. Non-flag args to pass to yc, typically
+        specifying an operation to perform (e.g. ['compute', 'images', 'list']
+        to list available images).
+    flags: OrderedDict mapping flag name string to flag value. Flags to pass to
+        yc (e.g. {'folder-id': 'my-folder-id'}). If a provided value is
+        True, the flag is passed to yc without a value. If a provided value
+        is a list, the flag is passed to yc multiple times, once with each
+        value in the list.
+    additional_flags: list of strings. Additional flags to append unmodified to
+        the end of the yc command (e.g. ['--metadata', 'color=red']).
+  """
+
+  def __init__(self, resource, *args):
```

Comment:
I suggest allowing for kwargs to make it easier to use:

```
def __init__(self, resource, *args, **kwargs):
  self.flags = kwargs
```

So that people can use it like `cmd = Yc('disk', size=100)` and do what is expected.